lib/std/json/hashmap.zig

```diff
@@ -0,0 +1,103 @@
+const std = @import("std");
+const Allocator = std.mem.Allocator;
+
+const ParseOptions = @import("static.zig").ParseOptions;
+const innerParse = @import("static.zig").innerParse;
+const innerParseFromValue = @import("static.zig").innerParseFromValue;
+const Value = @import("dynamic.zig").Value;
+const StringifyOptions = @import("stringify.zig").StringifyOptions;
+const stringify = @import("stringify.zig").stringify;
+const encodeJsonString = @import("stringify.zig").encodeJsonString;
+
+/// A thin wrapper around `std.StringArrayHashMapUnmanaged` that implements
+/// `jsonParse`, `jsonParseFromValue`, and `jsonStringify`.
+/// This is useful when your JSON schema has an object with arbitrary data keys
+/// instead of comptime-known struct field names.
+pub fn ArrayHashMapUnmanaged(comptime T: type) type {
+    return struct {
+        map: std.StringArrayHashMapUnmanaged(T) = .{},
+
+        pub fn deinit(self: *@This(), allocator: Allocator) void {
+            self.map.deinit(allocator);
+        }
+
+        pub fn jsonParse(allocator: Allocator, source: anytype, options: ParseOptions) !@This() {
```

Comment:
Consider the fully qualified name for this function:

`std.json.hashmap.ArrayHashMapUnmanaged.jsonParse`

Probably more ideal would be:

`std.json.ArrayHashMap.parse`

Reply:
this is implementing the interface expected by the std lib functions. the name is speaking to the `std.json` package, which is why `json` is in the name.